### ISSUE TITLE
Piper does not retain settings on Logi MX Master 3

### ISSUE THREAD

**Issue by rafikiphoto** [libratbag/piper#547]
Buttons can be set and work but as soon as Linux Mint 19.innamon is rebooted or sleeps the settings are lost. Piper 0.5.1.

**Comment by oleite** [comment]
I'm having the exact same issue here. Using MX Anywhere 2S, Ubuntu 20.04, Piper 0.5.1 .

**Comment by ikhsanalatsary** [comment]
having a problem when trying to set a button on MX master 2s on ubuntu 18.04
![Screenshot_2020-07-06_19-24-43](https://user-images.githubusercontent.com/5897415/86592973-956f1d00-bfbe-11ea-9e41-a61fc536bcf6.png)

**Comment by kkga** [comment]
Having the same issue with MX Master 3

**Comment by FFY00** [comment]
Unfortunately, the MX Master 3 does not save the settings between power cycles. Reapplying settings when a device is detected would be a reasonable feature request for piper.

I don't really have the time right now to implement it, maybe @stephanlachnit has?

**Comment by stephanlachnit** [comment]
> I don't really have the time right now to implement it, maybe @stephanlachnit has?

Exams are this months so no major code from me right now, maybe I'll look into it after the exams.

**Comment by rafikiphoto** [comment]
Thank you. Whenever time allows we'd all be grateful, I'm sure.

**Comment by SirNapkin1334** [comment]
Yes, this happens to me too. Except that there's no error message. When the system is restarted, all settings (except for RBG) are silently reset to default state. Logitech g502 w/ Arch Linux.

**Comment by FFY00** [comment]
That is weird, the G502 should work. Please open a new issue.

**Comment by stephanlachnit** [comment]
Just for reference (not sure if related), for me (G Pro) the polling rate isn't stored after a reboot / reconnect, I wrote about it some time ago in https://github.com/libratbag/piper/pull/494.

**Comment by rafikiphoto** [comment]
just a note to report I just upgraded to v20 Ulyana but the problem persists.
